app/src/main/java/com/babylon/wallet/android/presentation/settings/accountsecurity/accountrecoveryscan/scan/AccountRecoveryScanNav.kt

```diff
@@ -0,0 +1,95 @@
+package com.babylon.wallet.android.presentation.settings.accountsecurity.accountrecoveryscan.scan
+
+import androidx.annotation.VisibleForTesting
+import androidx.compose.animation.AnimatedContentTransitionScope
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavController
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavType
+import androidx.navigation.compose.composable
+import androidx.navigation.navArgument
+import com.babylon.wallet.android.presentation.navigation.markAsHighPriority
+
+@VisibleForTesting
+internal const val ARG_FACTOR_SOURCE_ID = "arg_factor_source_id"
+internal const val ARG_MNEMONIC = "arg_mnemonic"
```

Comment:
not sure this is the best idea to pass mnemonic like this

Reply:
Yes I agree with what Jakub says. We should be avoiding passing mnemonic in arguments.